strawberryfields/backends/tfbackend/circuit.py

```diff
@@ -888,7 +888,7 @@ def measure_homodyne(self, phi, mode, select=None, **kwargs):
 
             self._update_state(new_state)
 
-        return meas_result
+        return tf.cast(meas_result, dtype=ops.def_type)
```

Comment:
Should we update the functions that return the state tensor to do this casting as well?

Reply:
Potentially; lets keep this in mind when we review #323.